concordium-sdk/src/main/java/com/concordium/sdk/transactions/BlockItem.java

```diff
@@ -40,4 +57,28 @@ byte[] getBytes() {
         buffer.put(accountTransactionBytes);
         return buffer.array();
     }
+
+    public static BlockItem fromBytes(ByteBuffer source) {
+        val kind = BlockItemType.fromBytes(source);
+        switch (kind) {
+            case ACCOUNT_TRANSACTION:
+                val at = AccountTransaction.fromBytes(source);
+                return new BlockItem(at);
+            case CREDENTIAL_DEPLOYMENT:
+                throw new UnsupportedOperationException("Only account transactions are supported in this version. Credential deployments are not.");
+            case UPDATE_INSTRUCTION:
+                throw new UnsupportedOperationException("Only account transactions are supported in this version. Update instructions are not.");
+            default:
+                throw new UnsupportedOperationException("Unrecognized block item type.");
+        }
+    }
+
+    public static BlockItem fromVersionedBytes(ByteBuffer source) {
+        byte tag = source.get();
+        if (tag == 0) {
```

Comment:
There's a constant `VERSION` at the top of the class.

Reply:
And for unclear reasons that constant is an int. But I use it now.